### ISSUE TITLE
UDP DHT + Tor level anonymity 

### ISSUE THREAD

**Issue by nerd36** [ZeroNetX/ZeroNet#168]
You can use ready UDP DHT code. For anonymity users can use Lokinet instead of Tor which uses UDP. Maybe, you can even get Lokinet sponsorship, although they are small startup, themselves. Maybe buying their token before introducing feature give you some profit

**Comment by mx5kevin** [comment]
> Maybe buying their token before introducing feature give you some profit

No! Service with self made currency=All them are SCAM! See: https://github.com/ZeroNetX/ZeroNet/issues/170

**Comment by slrslr** [comment]
@nerd36 I had problem using Lokinet along with already installed Wireguard VPN. Like the two interfaces can not work together & Lokinet possibly requires to rent exit server as free is very limiter or hard to find?

**Comment by mx5kevin** [comment]
> @nerd36 I had problem using Lokinet along with already installed Wireguard VPN. Like the two interfaces can not work together & Lokinet possibly requires to rent exit server as free is very limiter or hard to find?

Lokinet try to connect to the network with UDP protocol. Tor connecting the Tor network with TCP. Lokinet does not have working exit nodes. Lokinet can use hidden services. Its advantage compared to TOR the lokinet addresses working with other programs like download managers. The problem with both of them if the whole network easy can scanned like in Tor all nodes can blocked. Lokinet can tunnel  more protocols like TOR like TCP. Currently not a TOR alternative because not have working exit nodes. The network is fast, faster than TOR. Not possible to set up custom ports to connect to the network. Does not have bridges like TOR. And have limited network options, does not have editable options like in Tor with torrc. I've never heard of Lokinet coin, and I hope there won't be one. Does not have SOCKS mode to connect it with programs.

**Comment by yanmaani** [comment]
Re: LokiNet
- Using UDP is not a gain in and of itself; you have to explain what purpose it would serve. There's a reason Tor won't use UDP, which is that routing arbitrary UDP through your exit nodes has a tendency to be quite unpleasant. "Moar protocols" is not really a design strategy.
- Lokinet does not have anywhere the level of review as Tor.
- Both Tor and Lokinet require either application-level support or running it through some kind of proxy (e.g. SOCKS-ification), there's not really any way around this
- Lokinet was created to sell a cryptocurrency and is forever tainted by this

Re: DHT
- Kademlia can trivially be implemented over TCP with very minor changes.
- DHTs are fundamentally limited by Sybil resistance.

Vote to close the issue.
